graphql/websocket.go

```diff
@@ -0,0 +1,155 @@
+package graphql
+
+import (
+	"encoding/binary"
+	"encoding/json"
+	"fmt"
+	"time"
+)
+
+const (
+	webSocketMethod         = "websocket"
+	webSocketTypeConnInit   = "connection_init"
+	webSocketTypeConnAck    = "connection_ack"
+	webSocketTypeSubscribe  = "subscribe"
+	webSocketTypeNext       = "next"
+	webSocketTypeError      = "error"
+	webSocketTypeComplete   = "complete"
+	websocketConnAckTimeOut = time.Second * 30
+)
+
+// Close codes defined in RFC 6455, section 11.7.
+const (
+	closeNormalClosure    = 1000
+	closeNoStatusReceived = 1005
+)
+
+// The message types are defined in RFC 6455, section 11.8.
+const (
+	// textMessage denotes a text data message. The text message payload is
+	// interpreted as UTF-8 encoded text data.
+	textMessage = 1
+
+	// closeMessage denotes a close control message. The optional message
+	// payload contains a numeric code and text. Use the FormatCloseMessage
+	// function to format a close message payload.
+	closeMessage = 8
+)
+
+type webSocketSendMessage struct {
+	Payload *Request `json:"payload"`
+	Type    string   `json:"type"`
+}
+
+type webSocketReceiveMessage struct {
+	Type    string          `json:"type"`
+	Payload json.RawMessage `json:"payload"`
+}
+
+func (w *webSocketClient) sendInit() error {
+	connInit := webSocketSendMessage{
+		Type: webSocketTypeConnInit,
+	}
```

Comment:
So in our case we need to provide bearer token in `connection_init` message:
```
{
  "type": "connection_init",
  "payload": {
    "Authorization": "Bearer token"
  }
}
```
Would it be possible to provide extension point for this ?